hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -433,6 +437,20 @@ void setSnapshotSummary(Map<String, String> parameters, Snapshot currentSnapshot
     }
   }
 
+  private void setPartitionSpec(TableMetadata metadata, Map<String, String> parameters) {
+    parameters.remove(TableProperties.DEFAULT_PARTITION_SPEC);
+    if (metadata.spec() != null && metadata.spec().isPartitioned()) {
+      parameters.put(TableProperties.DEFAULT_PARTITION_SPEC, PartitionSpecParser.toJson(metadata.spec()));
```

Comment:
`PartitionSpecParser.toJson()` doesn't convert a source column id to its name. For example, here is an example. 
```
{"spec-id":0,"fields":[{"name":"data_bucket","transform":"bucket[16]","source-id":2,"field-id":1000}]}
```
It only shows the source id(2),  the column name is `data` as the following code shows. It'd be more user-friendly if we show the column name.
```
    Schema schema = new Schema(
        required(1, "id", Types.IntegerType.get(), "unique ID"),
        required(2, "data", Types.StringType.get())
    );
    PartitionSpec spec = PartitionSpec.builderFor(schema)
        .bucket("data", 16)
        .build();
```

Reply:
Would we need to store schema on HMS side to make sense of the ids?  Otherwise today we are depending on the naming convention.

Reply:
Schema has been stored in HMS, https://github.com/apache/iceberg/blob/116026b40a43e477f68a34f8eead831ef139bc60/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L247, but schema id is missing when it converts Iceberg schema to Hive table schema. It may make sense to replace the source-id with source-name.

Reply:
Added the source name in the json string.

Reply:
(Chat offline) 

Imo, I think instead of changing it, we should just store schema object then for users interested in separate field, its more accurate then Hive schema and a bit less hacky than changing the partition-spec format to add it.

Reply:
Added a new property `current-schema` to avoid adding column name into partition spec and sort order. It duplicates the schema with Hive format, but we think it's worth to do that. The Hive format loses important column info(id).

Reply:
Yea also i think Hive columns may not have exact Iceberg schema in some cases.  For instance, the type conversions are not exact 1-to-1 mapping.